utils.go

```diff
@@ -336,6 +293,12 @@ func CborReadHeaderBuf(br io.Reader, scratch []byte) (byte, uint64, error) {
 		return 0, 0, err
 	}
 
+	defer func() {
```

Comment:
Note that the defer in `CborReadHeaderBuf` was missed in #64  - it will need to be fixed if this PR is not accepted.